pkg/ngctl/cmd/check/check.go

```diff
@@ -0,0 +1,203 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package check
+
+import (
+	"bytes"
+	"context"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1"
+	"github.com/vesoft-inc/nebula-operator/pkg/label"
+	cmdutil "github.com/vesoft-inc/nebula-operator/pkg/ngctl/cmd/util"
+	"github.com/vesoft-inc/nebula-operator/pkg/ngctl/cmd/util/ignore"
+)
+
+var (
+	checkLong = templates.LongDesc(`
+		Check whether the specified nebula cluster resources are ready.`)
+
+	checkExample = templates.Examples(`
+		# check whether the specified nebula cluster is ready
+		ngctl check 
+		
+		# check specified nebula cluster pods
+		ngctl check pods --nebulacluster=nebula`)
+)
+
+const resourceType = "nebulacluster"
+
+type CheckOptions struct {
+	Namespace         string
+	NebulaClusterName string
+	ResourceType      string
+	AllNamespaces     bool
+
+	runtimeCli client.Client
+	genericclioptions.IOStreams
+}
+
+func NewCheckOptions(streams genericclioptions.IOStreams) *CheckOptions {
+	return &CheckOptions{
+		IOStreams: streams,
+	}
+}
+
+// NewCmdCheck returns a cobra command for check whether nebula cluster resources are ready
+func NewCmdCheck(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	o := NewCheckOptions(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:     "check",
+		Short:   "check whether nebula cluster resources are ready",
+		Long:    checkLong,
+		Example: checkExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, args))
+			cmdutil.CheckErr(o.Validate(cmd))
+			cmdutil.CheckErr(o.RunCheck())
+		},
+	}
+
+	f.AddFlags(cmd)
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *CheckOptions) Complete(f cmdutil.Factory, args []string) error {
+	var err error
+
+	if o.NebulaClusterName, err = f.GetNebulaClusterName(); err != nil && !cmdutil.IsErNotSpecified(err) {
+		return err
+	}
+
+	if o.Namespace, err = f.GetNamespace(); err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		o.ResourceType = args[0]
+	}
+
+	o.runtimeCli, err = f.ToRuntimeClient()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validate validates the provided options
+func (o *CheckOptions) Validate(cmd *cobra.Command) error {
+	if o.NebulaClusterName == "" {
+		return cmdutil.UsageErrorf(cmd, "using 'ngctl use' or '--nebulacluster' to set nebula cluster first.")
+	}
+
+	if o.ResourceType == "" {
+		o.ResourceType = resourceType
+	}
+
```

Comment:
It is need to check the `o.ResourceType` ?

Reply:
If user don't specify the resourceType, we can use "nebulacluster" as a default type